src/sentry/api/endpoints/group_current_release.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import absolute_import
+
+from rest_framework.response import Response
+
+import sentry_sdk
+from sentry.api.base import EnvironmentMixin
+from sentry.api.bases import GroupEndpoint
+from sentry.api.helpers.environments import get_environments
+from sentry.api.serializers import serialize
+from sentry.api.serializers.models.grouprelease import GroupReleaseWithStatsSerializer
+from sentry.models import GroupRelease, ReleaseEnvironment, ReleaseProject
+
+
+class GroupCurrentReleaseEndpoint(GroupEndpoint, EnvironmentMixin):
+    def _get_current_release(self, group, environments):
+        release_projects = ReleaseProject.objects.filter(project_id=group.project_id).values_list(
+            "release_id", flat=True
+        )
+
+        release_envs = ReleaseEnvironment.objects.filter(
+            release_id__in=release_projects, organization_id=group.project.organization_id,
+        )
+        if environments:
+            release_envs = release_envs.filter(environment_id__in=[env.id for env in environments])
+        release_envs = release_envs.order_by("-first_seen").values_list("release_id", flat=True)
+
+        group_releases = GroupRelease.objects.filter(
+            group_id=group.id, release_id=release_envs[:1],
+        )
+        if environments:
+            group_releases = group_releases.filter(
+                environment__in=[env.name for env in environments],
+            )
+        try:
+            return group_releases[0]
+        except IndexError:
+            return None
+
+    def get(self, request, group):
+        """Get the current release in the group's project.
+
+        Find the most recent release in the project associated with the issue
+        being viewed, regardless of whether the issue has been reported in that
+        release. (That is, the latest release in which the user might expect to
+        have seen the issue.) Then, if the issue has indeed been seen in that
+        release, provide a reference to it. If not, indicate so with a null
+        value for "current release".
```

Comment:
As long as we're messing with the API contract, I wouldn't mind changing the `"currentRelease": null` thing into something more meaningful as well.

Perhaps
```javascript
{
  "currentRelease": { /* always points at the actual release unless none exists */ },
  "issueSeenInCurrentRelease": false
}
```

Only tricky part is that we serialize a `GroupRelease` object to populate `"currentRelease"`, which of course doesn't exist in the "false" case. Maybe I'll find a good option if I play around with the `ReleaseEnvironment` model. It will probably make the database interaction more expensive overall, but luckily we don't need to care any more! 😜